python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+        # eperTask()
+        # crosstalkTask()
+        # persistenceTask()
+
+        # TODO: For each eotest task, find out what the standard raft testing does for the optional params.
+        i.e. many have optional params for gains, bias-frames etc - if we want bitwise identicallity then we
+        need to know what is typically provided to these tasks when the camera team runs this code.
+        This can probably be worked out from https://github.com/lsst-camera-dh/lcatr-harness
+        but it sounds like Jim Chiang doesn't recommend trying to do that.
+
+        Parameters
+        ----------
+        butler : daf.persistence.butler
+            Butler for the repo containg the eotest data to be used
+        run : string or int
+            Optional run number, to be used for repos containing multiple runs
+        """
+        self.log.info("Running eotest routines direct")
+        runs = butler.queryMetadata('raw', ['run'])
+        if isinstance(run, int):
+            run = str(run)
+        if len(runs) != 1 and run is None:  # lots found and we don't know which one to choose
+            raise RuntimeError("Butler query found %s for runs. eotest datasets must have a run numbers, and"
+                               " must specify which run to use if a respoitory contains multiple runs."%runs)
+        elif run is not None and run not in runs:  # Could be specifying one of many, or one of one here
+            raise RuntimeError("Butler query found %s for runs, but the run specified (%s) "
+                               "was not among them."%(runs, run))
+        elif run is None:  # we know it's OK now
+            run = butler.queryMetadata('raw', ['run'])[0]
+        del runs  # we have run defined now, so remove this to avoid potential confusion later
+
+        if not os.path.exists(self.config.eotestOutputPath):
+            os.mkdir(self.config.eotestOutputPath)
+
+        ccds = butler.queryMetadata('raw', ['ccd'])
+        imTypes = butler.queryMetadata('raw', ['imageType'])
+        testTypes = butler.queryMetadata('raw', ['testType'])
+
+        ################################
+        ################################
+        # Run the Fe55 task
+        if self.config.doFe55:
+            fe55TaskDataId = {'run': run, 'testType': 'FE55', 'imageType': 'FE55'}
+            self.log.info("Starting Fe55 pixel task")
+            for ccd in ccds:
+                if 'FE55' not in testTypes:  # TODO: Remove not to test functionality
+                    msg = "No Fe55 tests found. Available data: %s"%testTypes
+                    if self.config.requireAllEOTests:
+                        raise RuntimeError(msg)
+                    else:
+                        self.log.warn(msg + "\nSkipping Fe55 task")
+                        break
+                fe55Filenames = [butler.get('raw_filename', dataId={'visit': visit,
+                                                                    'ccd': ccd})[0][:-3]
```

Comment:
Why the `[:-3]`?

Reply:
I don't really know why, but the butler returns a list of filenames which looks like:
`['/path/to/file/filename1.fits[0]', '/path/to/file/filename2.fits[0]']`, so that seems to be necessary 😬